core/drand_beacon_test.go

```diff
@@ -44,13 +44,18 @@ func TestBeaconProcess_Stop(t *testing.T) {
 	defer cancel()
 
 	proc.Stop(ctx)
-	if closed, ok := <-proc.WaitExit(); !ok || !closed {
-		t.Fatal("Expecting to receive from exit channel")
-	}
-	if _, ok := <-proc.WaitExit(); ok {
-		t.Fatal("Expecting exit channel to be closed")
-	}
+	closed, ok := <-proc.WaitExit()
+	require.True(t, ok, "Expecting to receive from exit channel")
+	require.True(t, closed, "Expecting to receive from exit channel")
+
+	_, ok = <-proc.WaitExit()
+	require.Falsef(t, ok, "Expecting exit channel to be closed")
+
 	proc.Stop(ctx)
+	<-proc.WaitExit()
```

Comment:
how come there are two additional waits on exist after we check the channel is closed?

Reply:
I agree that since we're testing it's closed we could remove the WaitExit reads that will always return since the chan is closed.

Reply:
Good catch. I should have changed the test to read as: `proc.Stop, proc.WaitExit, dd.Stop, dd.WaitExit`. I thought I did but apparently I missed it. Fixed in the next commit.